src/main/java/seedu/address/model/residentroom/UniqueResidentRoomList.java

```diff
@@ -0,0 +1,149 @@
+package seedu.address.model.residentroom;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.resident.Resident;
+import seedu.address.model.resident.exceptions.DuplicateResidentException;
+import seedu.address.model.resident.exceptions.ResidentNotFoundException;
+import seedu.address.model.residentroom.exceptions.DuplicateResidentRoomException;
+import seedu.address.model.residentroom.exceptions.ResidentRoomNotFoundException;
+
+/**
+ * A list of residents that enforces uniqueness between its elements and does not allow nulls.
```

Comment:
Perhaps change the resident references in this file to resident rooms (both in comment and code)